### PR TITLE
Styling for Buttons

### LINKED ISSUE #2184: Allow styling configuration for buttons
**Describe the problem/challenge you have**
Buttons in Octant are currently styled based on CSS rules from the frontend.

**Describe the solution you'd like**
For a given button, add the ability to:

- choose between solid, outlined, flat (no outline) styles
- disable a button
- create a [block button](https://clarity.design/angular-components/button/#block)
- show a button as loading
- allow setting colors to red, green, blue (error, success, info)

### PR DESCRIPTION
**Pull request by ftovaro**
Signed-off-by: ftovaro <f.tovar12@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**
- Fixes #2184  and #2310

**Special notes for your reviewer**:
Example of styles and sizes:

image

Example of how to use it:
```
// Buttons with style
btn1 := component.NewButton("ButtonStyleOutline", action.Payload{}, component.WithButtonStyle(component.ButtonStyleOutline))
btn2 := component.NewButton("ButtonStyleSolid", action.Payload{}, component.WithButtonStyle(component.ButtonStyleSolid))
btn3 := component.NewButton("ButtonStyleFlat", action.Payload{}, component.WithButtonStyle(component.ButtonStyleFlat))

// Buttons with size
btn4 := component.NewButton("ButtonSizeLarge", action.Payload{}, component.WithButtonSize(component.ButtonSizeLarge))
btn5 := component.NewButton("ButtonSizeBlock", action.Payload{}, component.WithButtonSize(component.ButtonSizeBlock))

// Buttons with style
btn6 := component.NewButton("ButtonStatusSuccess", action.Payload{}, component.WithButtonStatus(component.ButtonStatusSuccess))
btn7 := component.NewButton("ButtonStatusInfo", action.Payload{}, component.WithButtonStatus(component.ButtonStatusInfo))
btn8 := component.NewButton("ButtonStatusDanger", action.Payload{}, component.WithButtonStatus(component.ButtonStatusDanger))
btn9 := component.NewButton("ButtonStatusDisabled", action.Payload{}, component.WithButtonStatus(component.ButtonStatusDisabled))
```

**Release note**:
```
release-note
```

### CODE REVIEW

File: web/src/app/modules/shared/components/presentation/button-group/button-group.component.ts [hunk 1]
```diff
@@ -40,22 +40,7 @@ export class ButtonGroupComponent extends AbstractViewComponent<ButtonGroupView>
...
-        }
-        if (button.config.modal) {
-          this.modalView = button.config.modal;
-          const modal = this.modalView as ModalView;
-          this.modalService.setState(modal.config.opened);
-        }
-      });
-    }
-  }
+  update() {}
```
Comment (ftovaro) [inline web/src/app/modules/shared/components/presentation/button-group/button-group.component.ts:43]: This is now managed by the button itself

**Comment by GuessWhoSamFoo** [comment]
All the examples listed needs to be in storybook.

**Comment by GuessWhoSamFoo** [comment]
Holding per https://kubernetes.slack.com/archives/CM37M9FCG/p1618336449111900

**Comment by ftovaro** [comment]
Currently working on using Clarity core components

**Pull request merged**